File: java/gov/nasa/gsfc/gmsec/api/field/I16Field.java
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file I16Field.java
 */

package gov.nasa.gsfc.gmsec.api.field;

import gov.nasa.gsfc.gmsec.api.jni.field.JNII16Field;


/**
 * Specialized Field class that can be used to retain a signed 16-bit value.
 */
public class I16Field extends Field
{
	/**
	 * This constructor is for internal GMSEC API use only.
	 * @param field A JNII16Field object.
	 */
	public I16Field(JNII16Field field)
	{
		setInternal(field);
	}


	/**
	 * Default constructor.
	 *
	 * @param name Name of the field.
	 * @param data Data to associate with the field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name is null, or contains an empty string.
	 */
	public I16Field(String name, short data) throws IllegalArgumentException
	{
		validateName(name);

		setInternal(new JNII16Field(name, data));
	}


	/**
	 * Copy constructor.
	 *
	 * @param other The other I16Field object to copy.
	 *
	 * @throws IllegalArgumentException Thrown if the given I16Field is null.
	 */
	public I16Field(I16Field other) throws IllegalArgumentException
	{
		if (other == null)
		{
			throw new IllegalArgumentException("I16Field object is null");
		}

		setInternal(new JNII16Field((JNII16Field) Field.getInternal(other)));
	}


	/**
	 * Returns the data stored within the I16Field.
	 *
	 * @return A short value.
	 */
	public short getValue()
	{
		JNII16Field jI16Field = (JNII16Field) getInternal();

		return jI16Field.getValue();
	}
}

File: csharp/gmsec4/dox/field/I16Field.cs
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API {

/// <summary>
/// Specialized Field class that can be used to store a signed 16-bit value.
/// </summary>

public class I16Field : Field {

  /// <summary>Creates a I16Field object with the given value</summary>
  ///
  /// <param name="name">The name of the field</param>
  /// <param name="value">The value to store in the field</param>
  ///
  /// <exception cref="GmsecException">If the name is null or contains an empty-string</exception>
  public I16Field(string name, short value) {
  }

  /// <summary>Copy constructor</summary>
  /// <param name="other">The other I16Field object to copy</param>
  public I16Field(I16Field other) {
  }

  /// <summary>Returns the value associated with the field</summary>
  ///
  /// <returns>The value associated with the field</returns>
  public short GetValue() {
  }

}

}
